pkg/cloudprovider/aws/suite_test.go

```diff
@@ -207,6 +207,29 @@ var _ = Describe("Allocation", func() {
 				}
 				Expect(nodeNames.Len()).To(Equal(2))
 			})
+			It("should not combine GPU and non-GPU workloads", func() {
```

Comment:
I don't think I agree with this test. We definitely want to binpack pods into spare CPU/Mem on GPU nodes. Further, we don't even want to be aware of what a GPU is, as there are many unknown custom resource providers (vendor specific asic, etc) If customers want to prevent this, they can configured taints and tolerations accordingly.

Reply:
I don't like the test either, but my concern was:

Pod A - needs a GPU
Pod B - doesn't need a GPU

The binpacker can't fit Pod A and B onto the same node, so it launches two nodes with GPUs to support them.  The end result is a node with a GPU when a potentially cheaper node would have sufficed.

Reply:
Agree that we want to catch that binpacking case, but I that's not what were testing here. We want to test that two pods, one which requires a GPU, doesn't result in two GPUs. If they can fit together on a single gpu node's resources, they should.

Reply:
Reworked to test the intent